timebot/timebot.go

```diff
@@ -3,25 +3,62 @@
 package timebot
 
 import (
-	"strings"
+	"fmt"
+	s "strings"
 	"time"
 )
 
-// ParseTime takes a string and returns time.Time in time.UTC
-//
-// FIXME: Currently, it doesn't validate PST/PDT
+func CheckDaylightSavingZone(text string) (daylightSavingZone bool) {
+
+	tzDb := map[string]string{
+		"PST": "America/Los_Angeles",
+		"PDT": "America/Los_Angeles",
+	}
+
+	var tzText string = ""
+
+	for k, v := range tzDb {
+		if s.Contains(text, k) {
+			tzText = v
+		}
+	}
+	if tzText == "" {
+		return true
+	}
+
+	const longForm2 = "2006-01-02 15:04 MST"
+	loc, err := time.LoadLocation(tzText)
+	if err != nil {
+		fmt.Println("wrong tzDb")
+	}
+
+	t, _ := time.ParseInLocation(longForm2, text, loc)
+	tString := t.Format("2006-01-02 15:04 MST")
+	fmt.Printf("input text: %s, ParseInLocation: %s\n", text, tString)
+	return text == tString
+}
+
 func ParseTime(text string) (time.Time, bool) {
+
+	tzToTimeGap := map[string]string{
+		"PST": "-0800",
+		"PDT": "-0700",
+		"KST": "+0900",
+	}
+
+	passCheck := CheckDaylightSavingZone(text)
+	fmt.Println("passCheck", passCheck)
+
 	const longForm = "2006-01-02 15:04 -0700"
 
-	text = strings.Replace(text, "PST", "-0800", 1)
-	text = strings.Replace(text, "PDT", "-0700", 1)
-	text = strings.Replace(text, "KST", "+0900", 1)
-	t, err := time.Parse(longForm, text)
+	for key, value := range tzToTimeGap {
+		text = s.Replace(text, key, value, 1)
+	}
 
-	if err != nil {
+	t, err := time.Parse(longForm, text)
+	if err != nil || !passCheck {
```

Comment:
여기서 두번째 `err == nil && !passCheck` 을 제가 잘 모르겠는데 이 조건은 언제 걸리나요? :thinking: 

변수명은 `isDaylightZone` 등이 좀더 직관적인 것 같습니다

```suggestion
	if err != nil || !isDaylightZone {
```

Reply:
아예 DaylightZone 이 아니거나, DaylightZone 이면서 ok 인 경우를 모아서 표현한 것입니다.

Reply:
KST면 daylightzone이 아니면서 err가 nil이 아닌데 문제가 없나요?

Reply:
제가 깃헙 풀리퀘 UI에 익숙치 않아 정신없네요. 옆에 계시면 좋겠단 생각이 가득입니다. 
경모님은 err == nil && !passCheck 라고 하셨는데 제 코드는 if err != nil || !passCheck 인 것이지요? err 가 발생했거나 Check 결과가 통과되지 못한경우에 false 로 리턴하라는 게 맞는것 같은데요. 
KST 면 daylightzone 이 아니라 passCheck 를 통과할거고, err 가 nil 이지 않은가요?

Reply:
passCheck 가 
1. DaylightZone 이 아닌 경우와
2. DaylightZone 이면서 문제가 아닌 경우 둘을 모두 체크하는 것이라 isDaylightZone 은 명칭이 맞지 않는 것 같습니다. passDaylightZone 정도면 괜찮으려나요?

Reply:
아 이제 이해했습니다. CheckDaylightZone 를 제가 잘못이해했습니다. 

일단 리팩토링은 나중에하고 우선 CI를 PASS하는데 초점을 두고 하면 될 것 같습니다.